ui/lib/components/ListAndDetail/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { useLocation } from 'react-router-dom'
+import React from 'react'
+
+export interface ListAndDetailProps {
+  ListComponent: React.FunctionComponent
+  DetailComponent: React.FunctionComponent<{ style: React.CSSProperties }>
+  detailPathMatcher: (path: string) => boolean
+}
+
+export default function ({
+  ListComponent,
+  DetailComponent,
+  detailPathMatcher,
+}: ListAndDetailProps) {
+  const location = useLocation()
+  console.log('rendered')
+  return (
+    <div
+      style={{
+        position: 'relative',
+      }}
+    >
+      <ListComponent />
+      {detailPathMatcher(location.pathname) && (
+        <DetailComponent
```

Comment:
How about use an extra `div` to wrap the `<DetailComponent/>`? for example:

```
<div className={xxx}>
  <DetailComponent/>
</div>
```

So we don't need to modify the DetailPage.

Reply:
Just worried that the introduction of a new div will lead to the failure of some styles related to the div level. And it seems there is no such problem at present,